ui/src/app/utils/leo-runtime-initializer.tsx

```diff
@@ -309,61 +308,61 @@ export class LeoRuntimeInitializer {
     if (this.pollAbortSignal && this.pollAbortSignal.aborted) {
       // We'll bail out early if an abort signal was triggered while waiting for the poll cycle.
       return this.reject(
-        new LeoRuntimeInitializationFailedError('Request was aborted', this.currentRuntime));
+        new LeoRuntimeInitializationFailedError('Request was aborted', currentRuntimeStore.get()));
     }
     if (Date.now() - this.initializeStartTime > this.overallTimeout) {
       return this.reject(
         new LeoRuntimeInitializationFailedError(
           `Initialization attempt took longer than the max time allowed (${this.overallTimeout}ms)`,
-          this.currentRuntime));
+          currentRuntimeStore.get()));
     }
 
     // Fetch the current runtime status, with some graceful error handling for NOT_FOUND response
     // and abort signals.
     try {
-      this.currentRuntime = await this.getRuntime();
-      this.onStatusUpdate(this.currentRuntime.status);
+      currentRuntimeStore.set(await this.getRuntime());
```

Comment:
I think a hook to handle this may work. I have been working on it this morning. It seems like this may be a clean solution to the problem. Right now the hook is using the `LeoRuntimeInitializer`, but it could eventually handle the entire runtime cycle.

Reply:
I am resolving this conversation as this code has changed